src/pages/Swap/index.tsx

```diff
@@ -196,14 +197,27 @@ export default function Swap() {
   const [approval, approveCallback] = useApproveCallbackFromTrade(trade /* allowedSlippage */)
 
   // check if user has gone through approval process, used to show two step buttons, reset on token change
-  const [approvalSubmitted, setApprovalSubmitted] = useState<boolean>(false)
+  const [approvalsSubmitted, setApprovalsSubmitted] = useState<boolean[]>([])
+
+  const currentTradeIndex = useMemo(() => {
+    if (allPlatformTrades) {
+      return allPlatformTrades.findIndex(
+        trade => trade && platformOverride && trade.platform.name === platformOverride.name
+      )
+    } else {
+      return 0
+    }
+  }, [platformOverride, allPlatformTrades])
 
   // mark when a user has submitted an approval, reset onTokenSelection for input field
   useEffect(() => {
     if (approval === ApprovalState.PENDING) {
-      setApprovalSubmitted(true)
+      const newArray = [...approvalsSubmitted]
+      newArray[currentTradeIndex] = true
+      setApprovalsSubmitted(newArray)
     }
-  }, [approval, approvalSubmitted])
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [approval])
```

Comment:
Doesn't this work with `approvalSubmitted`?

Reply:
well its just reliant on the approval change rest are not neccesacry and we can cause unwanted re renders when we change approvals submitted and have it as dependency at the same time